File: Assets/Scripts/Module/AssetRuntime/Asset/LoadedAsset.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
namespace AssetRuntime
{
    public class LoadedAsset : Loaded
    {
        LoadedBundle m_belongBundle;
        Object m_asset;
        public Object Asset => m_asset;

        protected override void OnUnload()
        {
            Resources.UnloadAsset(m_asset);
            m_belongBundle.TryUnload();
            m_asset = null;
        }

        public void Clear()
        {
            m_asset = null;
            m_belongBundle = null;
        }

        public void SetInfo(LoadedBundle bundle, UnityEngine.Object asset)
        {
            m_belongBundle = bundle;
            m_asset = asset;
        }
    }
}

File: Assets/.lpproj/Apis/Unity/CS/AssetRuntime/LoadedAsset.lua
---@class CS.AssetRuntime.LoadedAsset : CS.AssetRuntime.Loaded
CS.AssetRuntime.LoadedAsset = {}

---@property readonly CS.AssetRuntime.LoadedAsset.Asset : CS.UnityEngine.Object
CS.AssetRuntime.LoadedAsset.Asset = nil

---@return CS.AssetRuntime.LoadedAsset
function CS.AssetRuntime.LoadedAsset()
end

function CS.AssetRuntime.LoadedAsset:Clear()
end

---@param bundle : CS.AssetRuntime.LoadedBundle
---@param asset : CS.UnityEngine.Object
function CS.AssetRuntime.LoadedAsset:SetInfo(bundle, asset)
end